src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -374,6 +392,11 @@ private void runFafReplayFile(Path path) throws IOException {
     Map<String, Integer> modVersions = replayInfo.getFeaturedModVersions();
     String mapName = replayInfo.getMapname();
 
+    // For some reason in the coop replay the map name is null in the metadata
+    // So we just take it directly from the replay data.
+    if (gameType.equals("coop"))
+      mapName = parseCoopMapName(rawReplayBytes);
```

Comment:
THAT IS AWESOME. Can we do that for generated maps pls? There the map is null aswell due to having no ID in the db.

Reply:
I'll need some sample of generated map to check that it can correctly parse the map name.

Reply:
Rebase onto develop and you will be able to generate maps in the create game dialog

Reply:
It's seems to be able to extract the map name (folder name in reality).
But are those uploaded to the vault ?

Here a link from a test that I did, but it doesn't exists.

http://content.faforever.com/faf/vault/maps/neroxismapgenerator_0.1.1_6524124518669183714.zip